### ISSUE TITLE
File created without a name

### ISSUE THREAD

**Issue by almayeen** [jaxon-php/jaxon-upload#2]
Hi everyone,

I have installed jaxon-core and jaxon-upload by the composer as in the below:
composer.json
"jaxon-php/jaxon-core": "~4.0"
"jaxon-php/jaxon-upload": "dev-main"

In the configuration, I enabled the upload and configured the upload directory.

My issue is when I'm using the upload feature, the file is created in the 'upload.default.dir' but without a name, for example:
9a9e83a43e7b67\.xls!!!!

Please guide me on how I can fix this issue.

I have noted that jaxon added a prefix in the javascript for each registered function! is that normal?

Thanks in advance.

**Comment by almayeen** [comment]
Finally, I figured out how to solve this issue:

Replace this line in class File function fromHttpFile
$xFile->sPath = $sUploadDir . $xFile->sName . '.' . $xFile->sExtension;
to
$xFile->sPath = $sUploadDir . $xFile->sFilename;

or set $xFile->sName to the proper value in that function because it's null

Best Regards

**Comment by feuzeu** [comment]
Hi,

The uploaded file name is randomly generated by Jaxon.
If you need to customize the name, you can set a sanitizer like this.

```php
jaxon()->upload()->sanitizer(function($sFileName, $sFieldName, $sFieldId) {
    // Return the name you want for the file here
    // $sFileName is the name of the uploaded file
    // $sFieldName is the value of the "name" attribute in the HTML input field
    // $sFieldId is the value of the "id" attribute in the HTML input field
});
```

I just realize this information is missing in the online doc. I will add it asap. Sorry.

**Comment by feuzeu** [comment]
You can change the prefix of the generated js classes using this option.

```php
jaxon()->setOption('core.prefix.class', '');
```
By default, it is set to `Jaxon`.

**Comment by almayeen** [comment]
Hi Feuzeu,
Thanks a lot for your reply and the great work.

I think the name of the directory is randomly generated like 9a9e83a43e7b67/.xls,
please check the body of 'fromHttpFile' function:

https://github.com/jaxon-php/jaxon-upload/blob/a8ef97ea602780039a53b05c9eb4c1b9b5b9a017/src/Manager/File.php#L83-L93

the value of the property $xFile->sName will be null at the end of this function, and this was the issue.

Best Regards,
Hussein

**Comment by feuzeu** [comment]
Hi,

This issue is fixed in the [1.0.1 release](https://github.com/jaxon-php/jaxon-upload/releases/tag/v1.0.1).
